### PR TITLE
feat: better Home Assistant MQTT structure and data

### PR DESCRIPTION
**Pull request by unreadablename**
#### Group Inverter data as device

Possibly breaking change for HA users

P. S. as bonus now it's easier to handle multiple inverters in HA #97 

Знімок екрана 2022-11-26 о 14 17 08

Знімок екрана 2022-11-26 о 14 17 34

### CODE REVIEW

**Comment by celsworth** [comment]
Thanks! Looks good.

I'm not too concerned with it being a breaking change at this stage, we're pre-1.0 for exactly that reason. I will put a note in the release notes that HA users should take extra care to check things still work though.

File: src/home_assistant.rs [hunk 1]
```diff
@@ -273,4 +287,16 @@ impl Config {
...
+    fn device(inverter: &config::Inverter) -> ConfigDevice {
+        let device = ConfigDevice {
+            identifiers: [
+                format!("lxp_{}", inverter.datalog()),
+            ],
+            manufacturer: "LuxPower".to_owned(),
+            name: format!("lxp_{}", inverter.datalog()),
+        };
+
+        return device;
```
Comment (celsworth) [inline src/home_assistant.rs:300]:
```suggestion
        ConfigDevice {
            identifiers: [
                format!("lxp_{}", inverter.datalog()),
            ],
            manufacturer: "LuxPower".to_owned(),
            name: format!("lxp_{}", inverter.datalog()),
```

**Comment by unreadablename** [comment]
@celsworth I'm going to add more useful registers to HA and later pack it as Home Assistant Add-on
btw, I'm very new to rust, so I will be very grateful for some comments and advices

**Comment by celsworth** [comment]
This looks good, it's a nice simple change to get started with :) I'll fix the tests, they're only failing because the returned JSON is different now.

**Comment by unreadablename** [comment]
@celsworth Thanks a lot

Is there any way to get inverter model from Inverter?

**Comment by celsworth** [comment]
No I don't think so :( There might possibly be a register for firmware version, I'd have to try reading it and see if it matches what I expect. It can be a future improvement though, this PR is probably fine to go in as it is if it makes HA more useful already.

**Comment by chriscn** [comment]
If this is a breaking change, it should not break the Energy history.

**Comment by unreadablename** [comment]
@chriscn unique id of sensors remains the same, so I guess it shouldn't break. But it worth to doublecheck

**Comment by chriscn** [comment]
> @chriscn unique id of sensors remains the same, so I guess it shouldn't break. But it worth to doublecheck

I'll try and do some testing in a staging instance and let you know.

**Comment by celsworth** [comment]
Thanks @chriscn . I'll merge this soon but hold off a release until we're confident it doesn't break anything it doesn't need to.

**Comment by celsworth** [comment]
This is in and I've fixed up tests. Added some initial details to the Wiki at https://github.com/celsworth/lxp-bridge/wiki/MQTT-Home-Assistant#update-from-v010

I can flesh it out a bit more once we're happy with this change, if anyone would care to report back about this.

One thing I did spot is these new message topics are retained but that leaves the older ones retained still. My broker still had a couple from months back when I was initially testing this! I don't know if HA will care about those; or how much we should really care about cleaning them up. I guess it was already possible to leave behind stale retained messages if you changed the configuration to remove a sensor being sent to HA anyway.

If this is a real problem I guess we could send a retain removal for any sensor we're not actively pushing to HA..

**Pull request merged**